src/librustc_resolve/lifetimes.rs

```diff
@@ -2879,19 +2909,59 @@ fn add_missing_lifetime_specifiers_label(
     count: usize,
     lifetime_names: &FxHashSet<ast::Ident>,
     snippet: Option<&str>,
+    missing_named_lifetime_spots: &[&hir::Generics<'_>],
 ) {
```

Comment:
Could you move this function into `diagnostics.rs`?

Reply:
done